src/metatensor/models/soap_bpnn/train.py

```diff
@@ -15,25 +22,19 @@
 logger = logging.getLogger(__name__)
 
 
-def train(train_dataset, hypers=DEFAULT_HYPERS, output_dir="."):
-    if len(train_dataset.targets) > 1:
-        raise ValueError(
-            f"`train_dataset` contains {len(train_dataset.targets)} targets but we "
-            "currently only support a single target value!"
-        )
-    else:
-        target_name = list(train_dataset.targets.keys())[0]
-
-    # Set the model's capabilities:
-    model_capabilities = ModelCapabilities(
-        length_unit="Angstrom",
-        species=train_dataset.all_species,
-        outputs={
-            target_name: ModelOutput(
-                quantity="energy",
-                unit="eV",
-            )
-        },
+def train(
+    train_datasets: List[Dataset],
+    validation_datasets: List[Dataset],
+    model_capabilities: ModelCapabilities,
+    hypers: Dict = DEFAULT_HYPERS,
+    output_dir: str = ".",
+):
+    # Perform canonical checks on the datasets:
+    logger.info("Checking datasets for consistency")
+    check_datasets(
+        train_datasets,
+        validation_datasets,
+        model_capabilities,
```

Comment:
I like this function!

I think this should be called in the `train` script. And also should include the test_datasets. This is something every model has to do.